### ISSUE TITLE
[BUG] 3005 onedir breaks salt-ssh priv: agent-forwarding -- environment doesn't contain SSH_AUTH_SOCK

### ISSUE THREAD

**Issue by jraby** [saltstack/salt#62796]
**Description**
Trying to use `salt-ssh` with a roster file specifying `priv: agent-forwarding` doesn't work when salt-ssh is installed from `onedir` (installed following https://docs.saltproject.io/salt/install-guide/en/latest/topics/install-by-operating-system/ubuntu.html)

Here's the error message:
```
    retcode:
        254
    stderr:
        Private key file need passphrase
    stdout:
```

From what I could gather, it seems that all the subprocesses created by salt do not have any environment variable set, which breaks agent forwarding, since it relies on `SSH_AUTH_SOCK` in order to find the agent's socket.

Here's a sample strace output showing the problem:

```
7160  execve("/opt/saltstack/salt/run/run", ["/opt/saltstack/salt/run/run", "ssh", "house", "test.ping"], ["SHELL=/bin/bash",  "TERM_PROGRAM_VERSION=3.2a",  "TMUX=/tmp/tmux-1000/default,1536"..., "SSH_AUTH_SOCK=/tmp/ssh-XXXX6FPB0"..., "EDITOR=vi", .....]) = 0
[...]

7164  execve("/bin/ssh", ["ssh", "-V"], ["LD_LIBRARY_PATH="] <unfinished ...>
7160  <... read resumed>"", 50000)      = 0
7164  <... execve resumed>)

7166  execve("/bin/ssh", ["ssh", "house.hostname", "-o", "KbdInteractiveAuthentication=no", "-o", "PasswordAuthentication=no", "-o", "GSSAPIAuthentication=no", "-o", "ConnectTimeout=65", "-o", "Port=22", "-o", "User=jean", "/bin/sh  << 'EOF'\nset -e\nset -u\n"...], ["LD_LIBRARY_PATH="] <unfinished ...>
7165  <... read resumed>"", 50000)      = 0
7166  <... execve resumed>)             = 0
```
I haven't yet tracked down where this empty environment comes from.

**Comment by OrangeDog** [comment]
Might be fixed by #62630

**Comment by jraby** [comment]
sounds like it probably does, are there nightly packages one could try to test this fix? Or running from source is the only way to test it?

**Comment by jraby** [comment]
fixed in 3005.1
